nessie/src/test/java/org/apache/iceberg/nessie/TestBranchVisibility.java

```diff
@@ -443,4 +443,26 @@ public void testWithRefAndHash() throws NessieConflictException, NessieNotFoundE
     nessieCatalog.createTable(identifier2, schema);
     Assertions.assertThat(nessieCatalog.listTables(namespace)).hasSize(2);
   }
+
+  @Test
+  public void testDifferentTableSameName() throws NessieConflictException, NessieNotFoundException {
+    String branch1 = "branch1";
+    String branch2 = "branch2";
+    createBranch(branch1, null);
+    createBranch(branch2, null);
+    Schema schema1 = new Schema(Types.StructType.of(required(1, "id", Types.LongType.get())).fields());
+    Schema schema2 = new Schema(Types.StructType.of(
+        required(1, "file_count", Types.IntegerType.get()),
+        required(2, "record_count", Types.LongType.get())).fields());
+
+    TableIdentifier identifier = TableIdentifier.of("db1.", "table1");
+
+    NessieCatalog nessieCatalog = initCatalog(branch1);
+    String location1 = nessieCatalog.createTable(identifier, schema1).location();
```

Comment:
can we also verify that each table has the correct schema?

Reply:
I didn't add it initially as it is a base functionality and it is already validated in some other test suite.

added now.